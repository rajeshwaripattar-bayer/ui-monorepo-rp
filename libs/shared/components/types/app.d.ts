export interface Notification {
  open: boolean
  timeout?: number
  message: string
  dismissButtonLabel?: string
  actionButtonProps?: {
    label: string
    onAction: () => void
  }
}
