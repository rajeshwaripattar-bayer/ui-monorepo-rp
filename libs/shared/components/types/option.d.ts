export type Option = { value: string; text: string }
export type ApplyAllProps = { label?: string; applied?: boolean }
