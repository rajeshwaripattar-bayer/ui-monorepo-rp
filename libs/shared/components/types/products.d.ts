import { DiscDiscount, PriceValueAndCurrency } from '@gc/types'
import { Strategy } from './discounts'

export interface ProductWithPrice {
  canView?: boolean
  canOrder?: boolean
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  netQuantity: number
  totalDiscount: number
  discounts?: { programName: string; strategy: Strategy }[]
  discountedUnitPrice: number
  unitPrice: number
  subTotal: number
  warehouse: { value: string; text: string }
  statusCounts?: {
    status: string
    count: number
    isBold?: boolean
  }[]
}

export interface PriceAndDiscountSummary {
  grossPrice: number
  discount: number
  discountPercentage: number
  discounts: {
    name: string
    value: number
  }[]
  netPrice: number
  discretionaryDiscounts: DiscDiscount[]
  totalDiscretionaryDiscount: PriceValueAndCurrency
  grossPriceDiscretionaryDiscount: PriceValueAndCurrency
  netPriceDiscretionaryDiscount: PriceValueAndCurrency
}

export interface ProductsByCrop {
  crop: string
  products: ProductWithPrice[]
  avgPrice: number
  summary: PriceAndDiscountSummary
}

export interface QuantityUpdateRequest {
  quantity: string | number
  productWithPrice: ProductWithPrice
  onlyCache?: boolean
  contingencyType?: 'alert' | 'dialog'
}

export interface ReturnedProducts {
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  warehouseName?: string
  UOM: string
  masterOrderNumber?: string
}
export interface ReturnedProductsByCrop {
  crop: string
  products: ReturnedProducts[]
}

export type StockOrderProduct = {
  name: string
  warehouse: { value: string; text: string }
  quantity: number
  confirmedQuantity: number
  unconfirmedQuantity: number
}

export type StockOrderProductsByCrop = {
  crop: string
  count: number
  products: StockOrderProduct[]
}
