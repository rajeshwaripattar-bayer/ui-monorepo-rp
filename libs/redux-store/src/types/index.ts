import type { ExtractReducerState, EnhancedStore } from '@gc/types'
import { ccApi, middlewareApi, fgCcApi } from '../rtk-apis'
import type { AppState, CartState, FiltersState } from '../slices'

export type GlobalRootState = {
  app: AppState
  cart: CartState
  filters: FiltersState
  [ccApi.reducerPath]: ExtractReducerState<typeof ccApi.reducer>
  [fgCcApi.reducerPath]: ExtractReducerState<typeof fgCcApi.reducer>
  [middlewareApi.reducerPath]: ExtractReducerState<typeof middlewareApi.reducer>
}

export type GlobalAppStore = EnhancedStore<GlobalRootState>
export type GlobalAppDispatch = GlobalAppStore['dispatch']
