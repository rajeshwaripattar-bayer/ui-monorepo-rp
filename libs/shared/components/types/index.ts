export * from './app'
export * from './option'
export * from './products'
