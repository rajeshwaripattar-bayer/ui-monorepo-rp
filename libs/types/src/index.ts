export * from './accounts'
export * from './aem'
export * from './shared'
export * from './orderDetails'
export * from './stockOrder'
export * from './api'
export * from './app'
export * from './cart'
export * from './ccApi'
export * from './configData'
export * from './discounts'
export * from './domainDef'
export * from './farmers'
export * from './helpers'
export * from './locale'
export * from './orders'
export * from './products'
export * from './programs'
export * from './quoteDetails'
export * from './quotes'
export * from './redux'
export * from './deliveries'
export * from './selectedAccount'
export * from './user'
export * from './filter'
export * from './shipmentDeliveries'
export * from './fgCcApi'
export * from './addressInfo'

