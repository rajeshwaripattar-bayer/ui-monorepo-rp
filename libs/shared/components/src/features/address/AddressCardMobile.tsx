import { Card, CardContent, CardTitle, CardBody } from '@element/react-card'
import styles from './AddressCardMobile.module.scss'
import { Address } from './Address'
import { AddressInfo } from '@gc/types'
import { TypoSubtitle } from '@element/react-typography'
import { ReactNode, isValidElement } from 'react'
/* eslint-disable-next-line */
export interface AddressCardMobileProps {
  title: ReactNode | string
  address: {
    addressInfo: AddressInfo
    typographyType?: string
  }
}

export function AddressCardMobile(props: AddressCardMobileProps) {
  const { title, address } = props
  return (
    <Card variant='outlined' className={styles.card}>
      <CardContent>
        <CardTitle primaryText={isValidElement(title) ? title : <TypoSubtitle level={1}>{title}</TypoSubtitle>} />
        <CardBody typographyType={address.typographyType || 'body2'} className={styles.card_body}>
          <Address address={address.addressInfo as AddressInfo} typographyType={address.typographyType ?? undefined} />
        </CardBody>
      </CardContent>
    </Card>
  )
}

export default AddressCardMobile
