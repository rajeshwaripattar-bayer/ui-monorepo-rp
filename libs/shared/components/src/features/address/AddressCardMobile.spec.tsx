import { render } from '@testing-library/react'

import AddressCardMobile from './AddressCardMobile'
import { AddressInfo } from '@gc/types'
import { ReactNode } from 'react'
import { TypoDisplay } from '@element/react-typography'

describe('AddressCard', () => {
  let addressInfo: AddressInfo
  let title: ReactNode | string
  beforeEach(() => {
    title = 'Ship From'
    addressInfo = {
      addressee: 'CARL GREEN WAREHOUSE',
      line1: '2344 Test Address Dr',
      town: 'Kansas City',
      region: {
        countryIso: 'US',
        isocode: 'KS',
        isocodeShort: 'KS',
        name: 'Kansas'
      },
      postalCode: '66062'
    }
  })
  it('should render successfully', () => {
    const { baseElement } = render(<AddressCardMobile title={title} address={{ addressInfo }} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render title with typography of subtitle 1 by default', () => {
    const { container } = render(<AddressCardMobile title={title} address={{ addressInfo }} />)
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--subtitle1').length).toBe(1)
  })

  it('should render title with expected typography, when passed as reactNode', () => {
    title = <TypoDisplay level={3}>{title}</TypoDisplay>
    const { container } = render(<AddressCardMobile title={title} address={{ addressInfo }} />)
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--headline3').length).toBe(1)
  })

  it('should render address with typography of body 2 by default', () => {
    const { container } = render(<AddressCardMobile title={title} address={{ addressInfo }} />)
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--body2').length).toBe(1)
  })

  it('should render address with typography that is passed as the property', () => {
    const { container } = render(
      <AddressCardMobile title={title} address={{ addressInfo, typographyType: 'display4' }} />
    )
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--headline4').length).toBe(1)
  })
})
