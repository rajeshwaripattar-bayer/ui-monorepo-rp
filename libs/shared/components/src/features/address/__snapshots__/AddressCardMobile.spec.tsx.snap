// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCard should render successfully 1`] = `
<body>
  <div>
    <div
      class="lmnt lmnt-card mdc-card mdc-card--outlined lmnt-card--left-top lmnt-theme-surface-bg card"
      style="height: auto;"
    >
      <div
        class="lmnt mdc-card__content"
      >
        <div
          class="lmnt lmnt-card-title lmnt-card-title--top-placement"
        >
          <div
            class="lmnt-card-title__text"
          >
            <span
              class="mdc-typography lmnt mdc-typography--subtitle1"
            >
              Ship From
            </span>
          </div>
        </div>
        <div
          class="lmnt-card-body  mdc-typography--body2 lmnt-theme-surface-bg card_body"
        >
          <div>
            <span
              class="mdc-typography lmnt mdc-typography--body2 address"
            >
              <label>
                CARL GREEN WAREHOUSE
              </label>
              <label>
                2344 Test Address Dr
              </label>
              <label />
              <label>
                Kansas City, KS 66062
              </label>
              <label />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
